sechub-commons-archive/src/test/java/com/mercedesbenz/sechub/commons/archive/SafeArchiveInputStreamTest.java

```diff
@@ -0,0 +1,267 @@
+// SPDX-License-Identifier: MIT
+package com.mercedesbenz.sechub.commons.archive;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.*;
+import static org.junit.Assert.assertThrows;
+import static org.junit.jupiter.api.Assertions.assertDoesNotThrow;
+
+import java.io.*;
+import java.nio.file.Files;
+import java.time.Duration;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.stream.Stream;
+
+import org.apache.commons.compress.archivers.ArchiveEntry;
+import org.apache.commons.compress.archivers.ArchiveInputStream;
+import org.apache.commons.compress.archivers.zip.ZipArchiveEntry;
+import org.apache.commons.compress.archivers.zip.ZipArchiveInputStream;
+import org.apache.commons.compress.archivers.zip.ZipArchiveOutputStream;
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtensionContext;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.Arguments;
+import org.junit.jupiter.params.provider.ArgumentsProvider;
+import org.junit.jupiter.params.provider.ArgumentsSource;
+
+class SafeArchiveInputStreamTest {
+
+    final List<File> tempFiles = new ArrayList<>();
+
+    @AfterEach
+    void tearDown() throws IOException {
+        for (File file : tempFiles) {
+            Files.deleteIfExists(file.toPath());
+        }
+    }
+
+    @Test
+    void new_safe_archive_input_stream_has_expected_properties() throws IOException {
+        /* prepare */
+        FileSize maxFileSizeUncompressed = new FileSize("10KB");
+        long maxEntries = 10L;
+        FileSize entryFileSize = new FileSize("1KB");
+        long maxDirectoryDepth = 10L;
+        Duration timeout = Duration.ofSeconds(10L);
+
+        ArchiveExtractionContext properties = new ArchiveExtractionContext(maxFileSizeUncompressed, maxEntries, maxDirectoryDepth, timeout);
+        ArchiveInputStream<?> archiveInputStream = createZipArchiveInputStream(maxEntries, entryFileSize, maxDirectoryDepth);
+
+        /* execute */
+        SafeArchiveInputStream result = new SafeArchiveInputStream(archiveInputStream, properties);
+
+        /* test */
+        assertThat(result.getProperties(), is(properties));
```

Comment:
Please rename properties variable and getter accordingly after the Context class has been renamed, 

```suggestion
        assertThat(result.getConstraintData(), is(constraintData));
```